crypto/ed25519/chainkd/expanded_key.go

```diff
@@ -0,0 +1,102 @@
+// This is an extension to ed25519.Sign that is compatible with NaCl `crypto_sign`
+// function taking 64-byte expanded private key (where the left part is a pre-multiplied
+// scalar and the right part is "prefix" used for generating a nonce).
+//
+// Invariants:
+// 1) Expanded(PrivateKey).Sign() == PrivateKey.Sign()
+// 2) InnerSign(Expanded(PrivateKey)) == Sign(PrivateKey)
+package chainkd
+
+import (
+	"crypto"
+	"crypto/sha512"
+	"errors"
+	"io"
+	"strconv"
+
+	"chain/crypto/ed25519"
+	"chain/crypto/ed25519/internal/edwards25519"
+)
+
+const (
+	// ExpandedPrivateKeySize is the size, in bytes, of a "secret key" as defined in NaCl.
+	ExpandedPrivateKeySize = 64
+)
+
+// ExpandedPrivateKey is the type of NaCl secret keys. It implements crypto.Signer.
+type ExpandedPrivateKey []byte
```

Comment:
Good point. I kept it similar to definitions in ed25519, but i definitely would prefer a fixed-length array.